.github/workflows/rust.yml

```diff
@@ -11,6 +11,33 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  build-contracts:
+    name: Build Cairo 1 contracts
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Fetch from cache
+      uses: actions/cache@v3
+      id: cache-programs
+      with:
+        path: cairo_programs/cairo-1-contracts/*.casm
+        key: cairo-1-contracts-cache-${{ hashFiles( 'cairo_programs/cairo-1-contracts/*.cairo' ) }}
+        restore-keys: cairo-1-contracts-cache-
+
+    # This is not pretty, but we need `make` to see the compiled programs are
+    # actually newer than the sources, otherwise it will try to rebuild them
+    - name: Restore timestamps
+      uses: chetan/git-restore-mtime-action@v1
+
+    - name: Install cairo 1 compiler
+      run: make build-cairo-1-compiler
+
+    - name: Build contracts
+      run: make -j cairo_1_test_contracts
+
```

Comment:
Can this be done extending the matrix in the other job?

Reply:
The file extension and the compiler is different so I felt it was more clear to do it in a separate job (as there would be added subtasks anyway)

Reply:
That can be parameterized too. See for example the `run-cairo-release` job.